### PR TITLE
feat: add Google Cloud Storage support (GCS)

### PR DESCRIPTION
**Pull request by shadinaif**
**TL;DR -** This is to add support to GCS files backend by adding a new option (`gcs`) to `ORA2_FILEUPLOAD_BACKEND`

JIRA: ---

**What changed?**

- `ORA2_FILEUPLOAD_BACKEND` environment variable accepts a new value `gcs` in addition to the existing options (`s3`, `filesystem`, `swift`, and `django`)
- No changes to the requirements. It is assumed that you already installed `django-storages[google]` instead of the default `django-storages` which will include `google-cloud-storages`

**Important to know!**
Using GSC with ORA means that `edx-platform` is already configured to use GCS instead of S3
- It is assumed that you already installed `django-storages[google]` instead of the default `django-storages` which will include `google-cloud-storages`
- It is assumed that `DEFAULT_FILE_STORAGE` is set to `storages.backends.gcloud.GoogleCloudStorage` in edx-platform. Other configurations might be needed too
- It is assumed that `edx-platform` functions directly and seamlessly with GCS without a proxy like MinIO or boto3

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled~~ not needed
- [ ] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

Using tutor development:
- Prepare your GSC storage, make sure to have a dedicated bucket for file upload
- Configure `edx-platform` to use GCS, including the installation of django-storages[google]
- Configure ORA2 to use GCS by setting `ORA2_FILEUPLOAD_BACKEND` to `gcs`
- Configure the dedicated bucket by setting `FILE_UPLOAD_STORAGE_BUCKET_NAME`
- Test uploading files from ORA2 and previewing them
- Test previewing uploaded files in ora2 MFE
- Verify that files exist in the dedicated storage
- Test deleting files
- Verify that files were deleted from the dedicated storage

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @shadinaif! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by itsjeyd** [comment]
@shadinaif Thank you for this contribution! Let us know when the changes are ready for review.

**Comment by shadinaif** [comment]
Thank you for the reminder @itsjeyd . Yes, it's on production and working fine. I'll rebase and set to review soon

**Comment by itsjeyd** [comment]
@shadinaif Sounds good, thanks for the update.

**Comment by itsjeyd** [comment]
Hey @shadinaif, do you have any updates about when you'll be able to come back to this PR?

**Comment by shadinaif** [comment]
Thanks for the follow-up @itsjeyd , I'll get back to it and complete it on Tuesday, June 4, 2024 I promise. I've added it to my schedule. Sorry for the delay 🙏🏼

**Comment by itsjeyd** [comment]
Sounds good, thanks for the update @shadinaif.

**Comment by itsjeyd** [comment]
@shadinaif It looks like tests will need some further tweaking here. Could you have a look?

**Comment by itsjeyd** [comment]
Hey @shadinaif, thanks for working on the tests.

It looks like some of them are still failing. Will you be taking another look?

**Comment by shadinaif** [comment]
Thanks for the reminder @itsjeyd , I'll do in a couple of days 👍🏼

**Comment by itsjeyd** [comment]
@shadinaif Just checking in to see if you're still planning to continue working on this PR? If you don't think you'll get to it in the short term, it might make sense to close it for the time being.

File: tox.ini [hunk 1]
```diff
@@ -7,7 +7,7 @@ deps =
     django42: Django>=4.2,<5.0
 
 commands =
-    pytest
+    pytest {posargs}
```
Comment (shadinaif) [inline tox.ini:10]: this is not related to the PR, it just makes it easier to run test for a single file or object locally

**Comment by shadinaif** [comment]
@itsjeyd thank you for your patience on this. I've added the needed tests, and I also had to fix some linter errors not related to the PR. All green now

**Comment by itsjeyd** [comment]
@shadinaif Great! I'm marking this PR as ready for review now.

**Comment by feanil** [comment]
Added the team to the repo and tagged them for review.

**Review by pomegranited — CHANGES_REQUESTED** [review]
Hi @shadinaif , I'm having trouble getting Google Cloud credentials configured to test this change. Are you planning to submit a GCS integration PR to the platform? 

I found [documentation](https://django-storages.readthedocs.io/en/latest/backends/gcloud.html) for initializing the `GS_CREDENTIALS` setting, but since it requires initializing that setting as a `google.oauth2.service_account.Credentials` object, I can't configure it with yml, and would have to modify the platform's `envs/*.py` files.

And without platform integration, I don't think we should add it to ORA2 either. Open to discussion though, let me know?

**Comment by itsjeyd** [comment]
Thanks for picking up the review for this PR @pomegranited!

**Comment by itsjeyd** [comment]
Hey @shadinaif, just checking in to see if you're still planning to continue working on this PR?

**Comment by shadinaif** [comment]
Hi @itsjeyd , looks like I missed the notification for the review. I'll continue with it on the weekend. Thank you for the ping!

**Comment by itsjeyd** [comment]
Hi @shadinaif, it's been about a month without any activity here, so I'm closing this PR for now. Feel free to reopen it (or open a new one) when you're ready to come back to this work.
